stakepooldclient/stakepooldclient.go

```diff
@@ -664,3 +663,68 @@ func (s *StakepooldManager) GetStakeInfo() (*pb.GetStakeInfoResponse, error) {
 	}
 	return nil, errors.New("GetStakeInfo RPC failed on all stakepoold instances")
 }
+
+// comaprePubkeys runs the gRPC command WalletMasterPubKeys on all stakepoold
+// instances, performs fn on all key, value pairs of received maps, and returns
+// whether any are true. Returns an error if any gRPC commands fail.
+func (s *StakepooldManager) comparePubKeys(fn func(string, string) bool) (bool, error) {
+	for i, conn := range s.grpcConnections {
+		client := pb.NewStakepooldServiceClient(conn)
+		resp, err := client.WalletMasterPubKeys(context.Background(), &pb.WalletMasterPubKeysRequest{})
+		if err != nil {
+			return false, fmt.Errorf("GetMasterPubkeys gRPC failed on stakepoold instance %d: %v", i, err)
+		}
+		for _, v := range resp.MasterPubKeys {
+			if fn(v.Account, v.PubKey) {
+				return true, nil
```

Comment:
This return statement in this loop implies that the grpc request may likely not be made to ALL stakepoold instances. Not sure if this was intended, PR description says `After stakepools are confirmed running, loop through them sending a grpc request to perform rpc requests on their respective dcrd to get maps of master pubkeys to account`; which I read to mean perform the grpc request on ALL stakepoold instances.

It's also important to check all instances to confirm/verify that the wallets on each backend server were all restored from the same seed; i.e. checking that the master pub key is the same across all the backend server wallets.

Reply:
The first true is sufficient for what I use it for. Return true if a default xpub is different and return true if xpub is found. Once a true if found there is no need to continue, and if no trues then all accounts on all wallets have been iterated.

It is checking that the master pub key for the default account is the same across all servers.

Reply:
I wrote it like this to avoid iterating the map twice. Perhaps the complexity is not worth it?

Reply:
Don't really quite get you. What I mean to confirm is this:
Does this code retrieve the voting wallet xpub key from ALL stakepoold instances and confirm that the xpub key is the same across ALL stakepoold instances?

Reply:
Yes.

It should. Does it not? 😬